src/SSVOpenHexagon/Core/MenuGame.cpp

```diff
@@ -70,15 +58,70 @@ MenuGame::MenuGame(Steam::steam_manager& mSteamManager,
         };
     window.onRecreation += [this] { refreshCamera(); };
 
+    initMenus();
+    initInput();
+
     levelDataIds =
         assets.getLevelIdsByPack(assets.getPackInfos().at(packIdx).id);
-
     setIndex(0);
-    initMenus();
-    initInput();
 }
 
-void MenuGame::init(bool error)
+void MenuGame::loadCommandLineLevel(const std::string& level)
+{
+    // retrieve the level index location
+    int levelIndex{-1}, i;
+    const auto& p(assets.getPackInfos());
+    std::vector<std::string> levelsIDs;
+    std::string packID, levelID;
+    std::vector<string>::iterator it;
+
+    for(i = 0; i < int(p.size()); ++i)
```

Comment:
Unless I am misunderstanding how this works, `level` is just the plain name of a level (e.g., "apeirogon"), and then you attempt to find any level named "apeirogon" in any of the installed packs by doing some string manipulation on the level id. 

This seems OK, but it does mean that it is not possible to properly handle cases where two separate packs have a level with the same name. Have you given any thought to this?

Reply:
I originally required both a pack and level name to boot into a level.
But I found it to be rather tiring to require two values and thought mapmakers should already put some thought in not having a same named level.
On second thought I see not everybody can know all names so I will reimplement the first version